packages/story-editor/src/components/panels/shared/generalPageAdvancement.js

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import {
+  useCallback,
+  useEffect,
+  useState,
+  useMemo,
+  useDebouncedCallback,
+} from '@googleforcreators/react';
+import { __ } from '@googleforcreators/i18n';
+import { v4 as uuidv4 } from 'uuid';
+import { trackEvent } from '@googleforcreators/tracking';
+import {
+  NumericInput,
+  Text,
+  THEME_CONSTANTS,
+  Toggle,
+} from '@googleforcreators/design-system';
+import { clamp } from '@googleforcreators/units';
+import {
+  DEFAULT_PAGE_DURATION,
+  DEFAULT_AUTO_ADVANCE,
+} from '@googleforcreators/output';
+
+/**
+ * Internal dependencies
+ */
+import { Row, Switch } from '../../form';
+import { SimplePanel } from '../panel';
+import { inputContainerStyleOverride } from './styles';
+
+const SwitchRow = styled.div`
+  margin-bottom: 16px;
+`;
+
+const JustifyEndRow = styled(Row)`
+  justify-content: flex-end;
+  gap: 8px;
+`;
+
+const MutedText = styled(Text).attrs({
+  size: THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL,
+})`
+  color: ${({ theme }) => theme.colors.fg.secondary};
+`;
+
+const LabelText = styled(MutedText).attrs({
+  as: 'label',
+})``;
+
+const MIN_MAX = {
+  PAGE_DURATION: {
+    MIN: 1,
+    MAX: 20,
+  },
+};
+
+function GeneralPageAdvancementPanel({
+  pageDuration = DEFAULT_PAGE_DURATION,
+  autoAdvance = DEFAULT_AUTO_ADVANCE,
+  onUpdate,
+  hasOverride = false,
+  isOverridden = false,
+  children,
+  panelName,
+  ...rest
+}) {
+  const [duration, setDuration] = useState(pageDuration);
+  const [override, setOverride] = useState(isOverridden);
```

Comment:
It's a bit hard to read the `hasOverride`, `isOverridden`, and `override` all together.

Maybe something like `overrideDefaultValue`, `setOverrideDefaultValue` and `isOverridden` => `hasCustomValue`; `hasOverride` => `allowsOverriding`?

Not sure if that's very clear but currently it takes a bit time to understand what's going on :D

Reply:
Done in b19b91e.